fastlane/lib/fastlane/actions/download_dsyms.rb

```diff
@@ -34,6 +34,14 @@ def self.run(params)
         end
       end
 
+      def self.download(url)
+        uri = URI.parse(url)
+        http = Net::HTTP.new(uri.host, uri.port)
+        http.use_ssl = (uri.scheme == "https")
+        res = http.get(uri.request_uri)
+        res.body
+      end
```

Comment:
Yup, I'm seeing this mostly in the actions in `fastlane`, since `fastlane` doesn't bring in any higher-level HTTP tooling.

Those other quality-of-life items are good to be considering as well, I agree.